server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTable.java

```diff
@@ -76,28 +116,27 @@ protected Mappings getUserDomainMappingsInternal(String user, String domain) thr
     }
 
     private Optional<Mappings> retrieveMappings(String user, String domain) {
-        List<String> mappings = session.execute(select(MAPPING)
-                .from(TABLE_NAME)
-                .where(eq(USER, getFixedUser(user)))
-                .and(eq(DOMAIN, getFixedDomain(domain))))
-            .all()
-            .stream()
-            .map(row -> row.getString(MAPPING))
-            .collect(Collectors.toList());
+        List<String> mappings = executor.execute(retrieveMappingStatement.bind()
+            .setString(USER, getFixedUser(user))
+            .setString(DOMAIN, getFixedDomain(domain)))
+            .thenApply(resultSet -> cassandraUtils.convertToStream(resultSet)
+                .map(row -> row.getString(MAPPING))
+                .collect(Guavate.toImmutableList()))
+            .join();
 
         return MappingsImpl.fromCollection(mappings).toOptional();
     }
 
     @Override
     protected Map<String, Mappings> getAllMappingsInternal() throws RecipientRewriteTableException {
-        Map<String, Mappings> map = session.execute(select(USER, DOMAIN, MAPPING)
-            .from(TABLE_NAME))
-            .all()
-            .stream()
-            .map(row -> new UserMapping(row.getString(USER), row.getString(DOMAIN), row.getString(MAPPING)))
-            .collect(Collectors.toMap(UserMapping::asKey, 
-                    userMapping -> MappingsImpl.fromRawString(userMapping.getMapping()),
-                    (first, second) -> first.union(second)));
+        Map<String, Mappings> map = executor.execute(retrieveAllMappingsStatement.bind())
+            .thenApply(resultSet -> cassandraUtils.convertToStream(resultSet)
+                .map(row -> new UserMapping(row.getString(USER), row.getString(DOMAIN), row.getString(MAPPING)))
+                .collect(Guavate.toImmutableMap(
+                    UserMapping::asKey,
+                    UserMapping::toMapping,
+                    Mappings::union)))
+            .join();
         return map.isEmpty() ? null : map;
```

Comment:
What you @chibenwa ternary expression hatter, let a ternary expression survive in your PR !!!!
Are you ok today ?